providers/implementations/kem/ml_kem.c

```diff
@@ -19,214 +19,215 @@
 #include "prov/implementations.h"
 #include "prov/securitycheck.h"
 #include "prov/providercommon.h"
-#include "prov/mlkem.h"
+#include "prov/ml_kem.h"
 
-#define BUFSIZE 1000
-#if defined(NDEBUG) || defined(OPENSSL_NO_STDIO)
-/* TODO(ML-KEM) to remove or replace with TRACE */
-static void debug_print(char *fmt, ...)
-{
-}
-#else
-static void debug_print(char *fmt, ...)
-{
-    char out[BUFSIZE];
-    va_list argptr;
-
-    va_start(argptr, fmt);
-    vsnprintf(out, BUFSIZE, fmt, argptr);
-    va_end(argptr);
-    if (getenv("TEMPLATEKM"))
-        fprintf(stderr, "TEMPLATE_KM: %s", out);
-}
-#endif
+static OSSL_FUNC_kem_newctx_fn ml_kem_newctx;
+static OSSL_FUNC_kem_freectx_fn ml_kem_freectx;
+static OSSL_FUNC_kem_encapsulate_init_fn ml_kem_encapsulate_init;
+static OSSL_FUNC_kem_encapsulate_fn ml_kem_encapsulate;
+static OSSL_FUNC_kem_decapsulate_init_fn ml_kem_decapsulate_init;
+static OSSL_FUNC_kem_decapsulate_fn ml_kem_decapsulate;
+static OSSL_FUNC_kem_set_ctx_params_fn ml_kem_set_ctx_params;
+
+typedef const ML_KEM_VINFO *vinfo_t;
```

Comment:
We do not use pointer types. Do we really need this typedef? If it is really needed, could it be please at least called `vinfo_constptr_t`? But I would prefer if you just used `const ML_KEM_VINFO *` instead.